src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java

```diff
@@ -0,0 +1,138 @@
+                .lineNumber(updateRequest.getPhoneNumber().getLineNumber())
+                .build();
+
+        if (user.getPhoneNumber() != phoneNumber) {
+            this.validatePhoneNumber(id, phoneNumber);
+        }
+
+        if (!(user.getEmailAddress().equals(updateRequest.getEmailAddress()))) {
+            this.validateEmailAddress(id, updateRequest.getEmailAddress());
+        }
+
+        final List<AysRole> roles = this.checkExistingRolesAndGet(updateRequest.getRoleIds());
+
+        user.setFirstName(updateRequest.getFirstName());
+        user.setLastName(updateRequest.getLastName());
+        user.setEmailAddress(updateRequest.getEmailAddress());
+        user.setCity(updateRequest.getCity());
+        user.setPhoneNumber(phoneNumber);
+        user.setRoles(roles);
+
+        userSavePort.save(user);
+    }
+
+
+    /**
+     * Validates the uniqueness of the provided phone number.
+     * Checks if there is any existing user with the same phone number.
+     *
+     * @param id          The unique identifier of the user being updated.
+     * @param phoneNumber The phone number to be validated.
+     * @throws AysUserAlreadyExistsByPhoneNumberException if the phone number is already associated with another user.
+     */
+    private void validatePhoneNumber(String id, AysPhoneNumber phoneNumber) {
+        userReadPort.findByPhoneNumber(phoneNumber)
+                .filter(existingUser -> !existingUser.getId().equals(id))
+                .ifPresent(existingUser -> {
+                    throw new AysUserAlreadyExistsByPhoneNumberException(phoneNumber);
+                });
+    }
+
+
+    /**
+     * Validates the uniqueness of the provided email address.
+     * Checks if there is any existing user with the same email address.
+     *
+     * @param id           The unique identifier of the user being updated.
+     * @param emailAddress The email address to be validated.
+     * @throws AysUserAlreadyExistsByEmailException if the email address is already associated with another user.
+     */
+    private void validateEmailAddress(String id, String emailAddress) {
+        userReadPort.findByEmailAddress(emailAddress)
+                .filter(existingUser -> !existingUser.getId().equals(id))
+                .ifPresent(existingUser -> {
+                    throw new AysUserAlreadyExistsByEmailException(emailAddress);
+                });
+    }
+
+
+    /**
+     * Checks the existence of roles by their IDs and returns the corresponding role entities.
+     *
+     * @param roleIds The set of role IDs to be checked and retrieved.
+     * @return A list of {@link AysRole} objects corresponding to the provided role IDs.
+     * @throws AysRolesNotExistException if any of the provided role IDs do not exist.
+     */
+    private List<AysRole> checkExistingRolesAndGet(final Set<String> roleIds) {
+        final List<AysRole> roles = roleReadPort.findAllByIds(roleIds);
```

Comment:
Farklı kurumun rolünü veya aktif olmayan bir rolü atamaya çalışmamamız gerekiyor
```suggestion
        final List<AysRole> roles = roleReadPort.findAllByIds(roleIds).stream()
                .filter(AysRole::isActive)
                .filter(role -> identity.getInstitutionId().equals(role.getInstitution().getId()));
```